### PR TITLE
NestedModule support

### PR DESCRIPTION
**Pull request by edgarfgp**

### CODE REVIEW

File: src/Fabulous.AST/Widgets/EscapeHatch.fs [hunk 1]
```diff
@@ -7,23 +7,22 @@ open Fantomas.Core.SyntaxOak
...
-            )
+            WidgetBuilder<'T>(EscapeHatch.WidgetKey, EscapeHatch.Node.WithValue(node))
 
 [<Extension>]
 type EscapeHatchYieldExtensions =
     [<Extension>]
     static member inline Yield(_: CollectionBuilder<'parent, ModuleDecl>, x: BindingNode) : Content =
-        { Widgets = MutStackArray1.One(Ast.TopLevelBinding(Ast.EscapeHatch(x)).Compile()) }
\ No newline at end of file
+        { Widgets = MutStackArray1.One(Ast.TopLevelBinding(Ast.EscapeHatch(x)).Compile()) }
```
Comment (nojaf) [inline src/Fabulous.AST/Widgets/EscapeHatch.fs:28]: Please add a test for this as well. I'd like to see that in action.

File: src/Fabulous.AST.Tests/WidgetTests.fs [hunk 2]
```diff
@@ -70,6 +70,50 @@ let x = 12
         
 let x = 12
 
+"""
+
+    [<Test>]
```
Comment (nojaf) [inline src/Fabulous.AST.Tests/WidgetTests.fs:75]: I would also bite the bullet now and use separate files for the different widgets.

File: src/Fabulous.AST/Fabulous.AST.fsproj [hunk 3]
```diff
@@ -18,6 +18,7 @@
         <Compile Include="Widgets\ModuleOrNamespace.fs" />
         <Compile Include="Widgets\Oak.fs" />
         <Compile Include="Widgets\Let.fs" />
+        <Compile Include="Widgets\NestedModule.fs" />
```
Comment (nojaf) [inline src/Fabulous.AST/Fabulous.AST.fsproj:21]: Each file should have its test counterpart?

Comment (edgarfgp) [reply src/Fabulous.AST/Fabulous.AST.fsproj:21]: Yeah that is a good idea 👍🏻

**Pull request merged**